src/data_structures/clifford_tableau.rs

```diff
@@ -177,6 +185,49 @@ impl CliffordTableau {
             .collect::<Vec<_>>();
         self.pauli_columns = sorted_pauli_columns;
     }
+
+    pub fn visualize_tableaus(&self) {
+        println!("Clifford Tableau ({} qubits):", self.size());
+        print!("    ||");
+        for i in 0..self.size() {
+            print!(" X{} Z{}|", i + 1, i + 1);
+        }
+        println!();
+        print!("+/- ||");
+        for i in 0..self.signs().len() {
+            if self.signs().get(i).map(|b| *b) == Some(true) {
+                print!(" +");
+            } else {
+                print!(" -");
+            }
+            print!(" ");
+            if i % 2 != 0 {
+                print!("|");
+            }
+        }
+        println!();
+
+        for i in 0..self.size() {
```

Comment:
Here it's probably cleaner to loop using `self.pauli_columns.iter().enumerate()`